org.eclipse.scanning.connector.epics/src/org/eclipse/scanning/connector/epics/EpicsV4ConnectorService.java

```diff
@@ -131,18 +131,19 @@ public void subscribe(IMalcolmDevice<?> device, MalcolmMessage msg, IMalcolmList
 
 		try {
 			EpicsV4ClientMonitorRequester monitorRequester = new EpicsV4ClientMonitorRequester(listener, msg);
-			PvaClientChannel pvaChannel = pvaClient.createChannel(device.getName(),"pva");
-	        pvaChannel.issueConnect();
-	        Status status = pvaChannel.waitConnect(REQUEST_TIMEOUT);
-	        if(!status.isOK()) {
-			String errMEssage = "Failed to connect to device '" + device.getName() + "' (" + status.getType() + ": " + status.getMessage() + ")";
-			logger.error(errMEssage);
-			throw new Exception(errMEssage);
-	        }
+			PvaClientChannel pvaChannel = pvaClient.createChannel(device.getName(), "pva");
+			pvaChannel.issueConnect();
+			Status status = pvaChannel.waitConnect(REQUEST_TIMEOUT);
+			if (!status.isOK()) {
+				String errMEssage = "Failed to connect to device '" + device.getName() + "' (" + status.getType() + ": "
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Failed to connect to device '" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)